block/manager.go

```diff
@@ -503,42 +572,74 @@ func (m *Manager) SyncLoop(ctx context.Context, cancel context.CancelFunc) {
 		case <-daTicker.C:
 			m.sendNonBlockingSignalToRetrieveCh()
 		case <-blockTicker.C:
-			m.sendNonBlockingSignalToBlockStoreCh()
-		case blockEvent := <-m.blockInCh:
-			// Only validated blocks are sent to blockInCh, so we can safely assume that blockEvent.block is valid
-			block := blockEvent.Block
-			daHeight := blockEvent.DAHeight
-			blockHash := block.Hash().String()
-			blockHeight := block.Height()
-			m.logger.Debug("block body retrieved",
-				"height", blockHeight,
+			m.sendNonBlockingSignalToHeaderStoreCh()
+			m.sendNonBlockingSignalToDataStoreCh()
+		case headerEvent := <-m.headerInCh:
+			// Only validated headers are sent to headerInCh, so we can safely assume that headerEvent.header is valid
+			header := headerEvent.Header
+			daHeight := headerEvent.DAHeight
+			headerHash := header.Hash().String()
+			headerHeight := header.Height()
+			m.logger.Debug("header retrieved",
+				"height", headerHash,
 				"daHeight", daHeight,
-				"hash", blockHash,
+				"hash", headerHash,
 			)
-			if blockHeight <= m.store.Height() || m.blockCache.isSeen(blockHash) {
-				m.logger.Debug("block already seen", "height", blockHeight, "block hash", blockHash)
+			if headerHeight <= m.store.Height() || m.headerCache.isSeen(headerHash) {
+				m.logger.Debug("header already seen", "height", headerHeight, "block hash", headerHash)
 				continue
 			}
-			m.blockCache.setBlock(blockHeight, block)
+			m.headerCache.setHeader(headerHeight, header)
 
-			m.sendNonBlockingSignalToBlockStoreCh()
+			m.sendNonBlockingSignalToHeaderStoreCh()
 			m.sendNonBlockingSignalToRetrieveCh()
 
 			err := m.trySyncNextBlock(ctx, daHeight)
 			if err != nil {
 				m.logger.Info("failed to sync next block", "error", err)
 				continue
 			}
-			m.blockCache.setSeen(blockHash)
+			m.headerCache.setSeen(headerHash)
+		case dataEvent := <-m.dataInCh:
+			data := dataEvent.Data
+			daHeight := dataEvent.DAHeight
```

Comment:
`dataEvent.DAHeight` is set only in tests.

Reply:
for consistency with header event, kept it there. might be used later. no harm to keep it for now?